src/tree/EnvironmentTreeItem.ts

```diff
@@ -69,8 +70,13 @@ export class EnvironmentTreeItem extends AzureParentTreeItem implements IAzureRe
     }
 
     public get description(): string {
-        const linkedDesc: string = localize('linkedTag', '{0} (linked)', this.data.properties.sourceBranch);
-        return this.inWorkspace ? linkedDesc : this.data.properties.sourceBranch;
+        if (this.data.properties.status !== 'Ready') {
+            // if the environment isn't ready, the status has priority over displaying its linked
+            return localize('statusTag', '{0} ({1})', this.data.properties.sourceBranch, this.data.properties.status);
+        }
+
+        const linkedTag: string = localize('linkedTag', '{0} (linked)', this.data.properties.sourceBranch);
+        return this.inWorkspace ? linkedTag : this.data.properties.sourceBranch;
```

Comment:
> Also this doesn't auto-refresh yet... Polling anyone? 😐

> Do you think it's worth creating a separate issue for the polling stuff?

@wwlorey I think (and hope) he's just joking about the polling because I was giving him a hard time yesterday re polling lol. We generally take a manual-refresh approach for our tree views. Auto-refresh can be quite expensive and unnecessary. Can you imagine if a user had all of our extensions installed and they were all polling on slow internet? The value vs. cost just isn't there

Reply:
Eric is correct ^